packages/material-ui/src/Snackbar/Snackbar.js

```diff
@@ -127,9 +129,13 @@ const Snackbar = React.forwardRef(function Snackbar(props, ref) {
     ...other
   } = props;
 
+  const theme = useTheme();
   const timerAutoHide = React.useRef();
   const [exited, setExited] = React.useState(true);
 
+  const desktop = useMediaQuery(theme.breakpoints.up('sm'));
```

Comment:
q: What do you consider desktop? As far as I can tell tying desktop to the screen size is no longer accurate. It was maybe 5-10 years ago but now we have all sorts of display sizes that I wouldn't consider "desktop".

Reply:
>As far as I can tell tying desktop to the screen size is no longer accurate
that's true

since device judgement with no dependency will be too many work, i thought give users some space so that they can decide what they want to consider as desktop would work

like,
a. add `desktop` props for `Snackbar` component and just use that flag.
b. use custom break point inside `Snackbar`, when user set `desktop` on breakpoints, its value is used.
https://material-ui.com/customization/breakpoints/#custom-breakpoints

Reply:
@kodai3 Why are you deviating from the current tradeoff? Why using JavaScript when we can use the CSS media queries to change the position of the snackbar? Wouldn't we use CSS to better handle the initial paint?